src/core/border_router/routing_manager.hpp

```diff
@@ -540,6 +540,22 @@ class RoutingManager : public InstanceLocator
         mPdPrefixManager.ProcessPlatformGeneratedRa(aRouterAdvert, aLength);
     }
 
+    /**
+     * Handles a prefix delegated from a DHCPv6 PD server. The prefix is received on the DHCPv6 PD client callback and
+     * then this method can be used to configure the prefix in the Routing Manager module.
+     *
+     * Note: This method is a part of DHCPv6 PD support on Thread border routers. For platforms where it doesn't make
+     * sense to generate a RA to set a DHCPv6 PD prefix this method can be used to set the prefix directly. The lifetime
+     * of the prefix can be updated by calling the function again with updated values.
+     *
+     * @param[in] aPrefixInfo  A pointer to the prefix information received from the DHCPv6 PD server.
+     *
+     */
+    void ProcessDhcpPdPrefix(const PrefixTableEntry *aPrefixInfo)
```

Comment:
Declare `aPrefixInfo` as a reference (not a pointer)? (We already `AssertPointerIsNotNull(aPrefixInfo);` in the `routing_manager_api.cpp`).

```suggestion
    void ProcessDhcpPdPrefix(const PrefixTableEntry &aPrefixInfo)
```